sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs

```diff
@@ -29,7 +29,7 @@ public TestRunDto GetTestRun()
         {
             var startTime = _cloudRunMetadata.TestRunStartTime.ToString("yyyy-MM-ddTHH:mm:ssZ");
             var gitBasedRunName = ReporterUtils.GetRunName(CiInfoProvider.GetCIInfo())?.Trim();
-            string runName = string.IsNullOrEmpty(gitBasedRunName) ? _cloudRunMetadata.RunId! : gitBasedRunName!;
+            string runName = !string.IsNullOrEmpty(_cloudRunMetadata.RunName) ? _cloudRunMetadata.RunName! : string.IsNullOrEmpty(gitBasedRunName) ? _cloudRunMetadata.RunId! : gitBasedRunName!;
```

Comment:
Inverse the if logic. Also consider moving it to a if else condition for better readability. Also add tests for the same.

Reply:
+1 for increasing readability.
IN nutshell, we will give preference to user passed value when given